activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -29,8 +36,35 @@ def create_table(table_name, options = {})
             options[:auto_increment] = true
           end
 
-          super
+          if block_given?
+            super(table_name, options) do |t|
+              class << t
+                prepend TableDefinition
```

Comment:
`class << t` is the eigenclass of the `t` instance, this `prepend` is to the the eigenclass, not the superclass.